### DISCUSSION TITLE
decoding Epson RLC bug

### DISCUSSION THREAD

**Discussion by kdschlosser** [bengtmartensson/IrScrutinizer#440] [Decoding issues]
...

**Comment by bengtmartensson** [comment]
Hi Kevin,

The Epson protocol is not exactly widely used or very well known. It is to my knowledge first discussed [here](http://www.hifi-remote.com/forums/viewtopic.php?t=17051). From this thread, I wrote the IRP form

    {38.4k,577}<2,-1|1,-2|1,-1|2,-2>((4,-1,D:8,T1:2,OBC:6,T2:2,S:8,1,-75m)*,(4,-1,D:8,~F1:2,OBC:6,~F2:2,S:8,1,-250m)) [D:0..255,S:0..255,OBC:0..63,T1:0..3,T2:0..3,F1:0..3,F2:0..3]

This is quite an unusual protocol: it lacks an into sequence, and has a repeat- and an ending sequence, (where the latter is mandatory)  that are identical, except for the parameters being different. I do not know any other protocol with that property.

The decoding in IrpTransmogrifier (which is used in IrScrutinizer) can try to decode either an _IrSequence_, or an _IrSignal_ (already decomposed in intro-, repeat- and ending sequence). The decoding is pretty robust if given an IrSequence, but, if given an IrSignal, and the decomposition (in intro-, repeat- and ending-sequence) is not compatible with the IRP, the decoding result may not be as expected. So, for example

> In a real world application the RLC can be one of the 2 following RLC examples.
 
> ... 
     [[+2308, -577, +1154, -1154, +1154, -577, +1154, -1154, +1154, -577, +577, -1154, +577, -577, +1154, -577, +1154, -1154, +577, -577, +577, -1154, +577, -1154, +577, -577, +577, -1154, +577, -75000], [+2308, -577, +1154, -1154, +1154, -577, +1154, -1154, +1154, -577, +1154, -577, +577, -577, +1154, -577, +1154, -1154, +1154, -1154, +577, -1154, +577, -1154, +577, -577, +577, -1154, +577, -250000]]

The last IrSignal does not fit the scheme at all: The intro is required to be empty, but isn't. The ending is required to be non-empty, but isn't. So the decode is not what expected. (Strictly speaking, it should possibly be rejected altogether...) (If trying all the numbers as one IrSequence, it will decode to the expected parameters.)

I do nor recognize any decoding bugs in your text.

A "scrutinized" signal is independently tried for decoding and analyzing. The former uses a priori knowledge in the form of the IRP database, the latter uses no a priori knowledge. Therefore, the question

> So first question is what is it msb or lsb?

is as meaningless as the classical "is the glass half full or half empty?".

I am not  going to review you attempted hand-decode. However, I would recommend that you try to get acquainted to [invoking the analyzer of irptransmogrifier from the command line](http://www.harctoolbox.org/IrpTransmogrifier.html#analyze). (You already have it installed along with IrScrutinizer.)

**Comment by kdschlosser** [comment]
How is this a discussion?  There is a bug in the program as the  IR codes that are being generated DO NOT conform to what the IRP is. so either the IRP is wrong or the generated codes are wrong.


Gotta love when a problem is found and the report of the problem gets ignored.

**Comment by bengtmartensson** [comment]
I explained in my previous contribution that your problem is not a bug, but you appear not to have read it.
